server/views/serviceProviderReferrals/dashboard.njk

```diff
@@ -18,9 +18,9 @@
 
       {{ mojSubNavigation(subNavArgs) }}
 
-      {% if presenter.dashboardType === 'All open cases' %}
+      {% if (presenter.dashboardType === 'All open cases') or (presenter.dashboardType === 'Unassigned cases')  %}
```

Comment:
These fragments
```ts
{% if (presenter.dashboardType === 'All open cases') or (presenter.dashboardType === 'Unassigned cases')  %}
```
```ts
{% if (presenter.dashboardType === 'My cases') or (presenter.dashboardType === 'Completed cases')%}
```
```
There are no results for "{{presenter.SearchText | escape }}" in {{presenter.casesType}}
```
```
If you still cannot find it, it may not be in {{presenter.casesType}}. Check in other case lists.
```
look like they are all working around that we don't have two abstractions

- categorical dashboard types (`All open cases` and `open cases` as configuration seems wrong)
- the ability to decide if something is searchable

This seems to be corroborated by fragments like
```ts
  get hrefLinkForSearch(): string {
    if (this.dashboardType === 'All open cases') {
      return `/service-provider/dashboard/all-open-cases`
    }
    if (this.dashboardType === 'Unassigned cases') {
      return `/service-provider/dashboard/Unassigned-cases`
    }
    return ''
  }
```
where this could be properties of a class (where open, unassigned, my, completed could be separate instances)

What do you think?

Reply:
I agree with you @sldblog , I have changed the code like this